kloppy/domain/services/transformers/__init__.py

```diff
@@ -11,41 +11,86 @@
     Orientation,
     PitchDimensions,
     Point,
+    Point3D,
     Team,
     TrackingDataset,
+    CoordinateSystem,
+    Origin,
 )
 from kloppy.domain.models.event import Event
 
 
 class Transformer:
     def __init__(
         self,
-        from_pitch_dimensions: PitchDimensions,
-        from_orientation: Orientation,
-        to_pitch_dimensions: PitchDimensions,
-        to_orientation: Orientation,
+        from_coordinate_system: CoordinateSystem = None,
+        from_pitch_dimensions: PitchDimensions = None,
+        from_orientation: Orientation = None,
+        to_coordinate_system: CoordinateSystem = None,
+        to_pitch_dimensions: PitchDimensions = None,
+        to_orientation: Orientation = None,
     ):
-        self._from_pitch_dimensions = from_pitch_dimensions
+
+        print("hello")
+        if (
+            from_pitch_dimensions is not None
```

Comment:
`if from_pitch_dimensions and from_coordinate_system [etc]`

those variables shouldn't have any falsy value, other than `None`